### DISCUSSION TITLE
Color Schema not working correctly

### DISCUSSION THREAD

**Discussion by ModdyLP** [nunocoracao/blowfish#600] [Q&A] [answered]
Hay,

i dont know if it my fault, but after i updated the theme to the newest version, my color scheme is ignored. I looked into the compiled main.css and saw that there are some changes of colors from a variable to a static color. Maybe this is conflicting with the schema variables. 

Can someone test this and verify if this is a bug or my fault? Maybe I updated the files wrong.

**Comment by ModdyLP** [comment]
After this Commit: https://github.com/nunocoracao/blowfish/commit/4cefe3a252e2af17bc83005525a81234beaea782 the Rebuild causes the break maybe: https://github.com/nunocoracao/blowfish/commit/5b9e28503bedf9168f898d01f094af8eb533d4bb

**Reply by nunocoracao** [reply]
Will check

**Reply by ModdyLP** [reply]
I made a research and builded it fresh on my machine. The behavior persists. I found a issue in the tailwindcss repository: https://github.com/tailwindlabs/tailwindcss/issues/9143. Maybe this is relevant for us. I will test some things and if I find a solutions i will push this.

**Reply by JVimes** [reply]
5b9e285 seems promising. Looks like a color variable was evaluated when generating CSS:

![image](https://user-images.githubusercontent.com/6148201/225163235-e506ef1f-0e2a-407d-8049-541ecc867b90.png)

**Reply by JVimes** [reply]
@nunocoracao I think the beta1 version of **tw-elements** is the culprit:

- With tw-elements 1.0.0-beta1 I get `text-decoration-color: rgb(171 194 232 / 1);`
- With tw-elements 1.0.0-alpha13 I get `text-decoration-color: rgba(var(--color-primary-300), 1);`

**Reply by JVimes** [reply]
To help anyone in the meantime, [here's](https://gist.github.com/JVimes/cdbc012a91b3cf29c2cadbfd654ce23b) a `/generated/main.css` built from latest Blowfish (with old tw-elements) that seems to work.

**Reply by nunocoracao** [reply]
@ModdyLP @JVimes I didn't have time to look into this but I am running thus in Blowfish and my personal website with no issues. I wouldn't advise replacing that css file as it might cause future problems. Can I get more details on the problem, potentially a repo URL?

**Reply by ModdyLP** [reply]
@nunocoracao Sure, here is my repo with a branch where I build the update and pushed it. Simple start the example and then you will see that the fire schema is not applied and the compiled main.css looks wrong. https://github.com/ModdyLP/blowfish/tree/fix_Colors 

I am also investigating and looking for a fix, i teared it down to an update from tailwindcss in tailwind elements from version 3.0.7 to 3.2.4. Between these versions many fixes are done and the parsing was extended to support rgb and hsl codes. Maybe the parser is not working correctly. As i mentioned there is already a issue, that the styles vars do not apply if there is a alpha notation. I tried removing it and some other things, sometimes it works partly. I will try to fix it and provide it if i found something.

**Reply by nunocoracao** [reply]
OK found the problem

**Reply by nunocoracao** [reply]
issuing a fix soon
https://github.com/nunocoracao/blowfish/issues/611

**Comment by JVimes** [comment]
Me too. I don't think I see anything reading --color-primary- or --color-secondary colors other than chart.js. The button background color seems to be hard coded.

**Comment by nunocoracao** [comment, marked as answer]
fixed in https://github.com/nunocoracao/blowfish/issues/611

**Reply by ModdyLP** [reply]
Very Nice its working now, is this require not necessary anymore?

**Reply by nunocoracao** [reply]
never was... leftovers of when I was trying to figure out the breaking changes from tw alpha to beta. Tested the shortcodes that use it and everything seems to be ok.

**Reply by ModdyLP** [reply]
Ah, okay understandable. That was a line i never thought about. Because I thought it was necessary and didn't touch it.
